numpy/core/src/multiarray/sequence.c

```diff
@@ -38,7 +38,12 @@ array_contains(PyArrayObject *self, PyObject *el)
     if (res == NULL) {
         return -1;
     }
+
     any = PyArray_Any((PyArrayObject *)res, NPY_MAXDIMS, NULL);
+    if (any == NULL) {
+        return -1;
+    }
+
     Py_DECREF(res);
```

Comment:
The res should be decref'd before the error check. It is not needed anymore in both branches.

Reply:
That is too far up :). You have to hold on to the object (read: the reference that gets decref'd) as long as you are using it. And it is used in the `PyArray_Any` call.

Reply:
Ohh sorry, my bad, will fix it. Thanks.